tekton-proposal.md

```diff
@@ -0,0 +1,73 @@
+# Knative Build and Tekton Pipelines Steering Committee proposal
+
+Jason Hall ([jasonhall@google.com](mailto:jasonhall@google.com))
+
+Feb 25 2019
+
+## Objective
+
+Propose a resolution to the ongoing confusion and FUD around the future of Knative Build with regard to the recently announced Tekton Pipelines change. This doc should be used as a forum for external comment and questions, to resolve any concerns before the Steering Committee decides whether to adopt the proposal.
+
+Please leave comments on this PR. The text of the proposal may change based on feedback, and the PR will retain the history of these edits.
+
+## Background
+
+Knative Build was conceived at the beginning of the Knative project alongside the other serverless-focused components (Serving and Eventing) to enable a "serverless development experience". Its goal was to enable users to specify their source to a Knative Service deployment and the steps to take to make it a container, and deploy that container, while interacting with a single custom resource.
+
+From the beginning, Build had bigger ambitions. Its [2018 roadmap](https://github.com/knative/build/blob/master/roadmap-2018.md) mentions features that aren't directly related to furthering serverless deployments (triggering, workflow, etc.). These aren't accelerators for serverless deployments, these are CI/CD features!
+
+This tension has existed in Build from the beginning: did it exist to accelerate the Knative serverless platform? Or is it supposed to be a full-featured CI/CD platform? How should Build prioritize features that don't have anything to do with the rest of Knative? These are not hypothetical questions, these are questions we've heard from users, operators and contributors.
+
+To further add confusion, in August 2019 Knative began the Pipelines effort in the knative/build-pipeline repo. This was an experimental ground-up design to build on the experience with the Build resource, and build real CI/CD workflow infrastructure running on Kubernetes. Its [2019 roadmap](https://github.com/knative/build-pipeline/blob/master/roadmap-2019.md) (the knative/build repo never had a 2019 roadmap) included all the CI/CD features from 2018 and more, and makes only [passing mention](https://github.com/knative/build-pipeline/blob/master/roadmap-2019.md#dont-break-serving) of its relationship with Serving.
```

Comment:
My personal take would be to remove language that tend to exacerbate "tension" and "confusion" and instead just be very factual.

Reply:
I think the central thing we're trying to accomplish is to alleviate the tension and confusion we've caused, on two fronts:

1. confusion about the rollout and decision-making process for the change being proposed here, which we're trying to alleviate by being upfront in this proposal and in the public comment process for it. We should have done this from the beginning.

1. tension about the goals and roadmap for Knative Build (and Knative as a whole) from the outset. Knative has been "serverless" but also contributors in build-pipelines are planning CI/CD features. We're attempting to alleviate that tension by moving the CI/CD portions out to a new project.

If there are parts where I'm over-selling the tension, please let me know, but I think we have to be honest about the tension we've caused in order to resolve it.